sdk/deviceprovisioningservices/Microsoft.Azure.Management.DeviceProvisioningServices/tests/ScenarioTests/DeviceProvisioningClientTest.cs

```diff
@@ -1,153 +1,165 @@
+            if (availabilityInfo.NameAvailable.HasValue && !availabilityInfo.NameAvailable.Value)
+            {
+                //it exists, so test the delete
+                await _provisioningClient.IotDpsResource
+                    .DeleteAsync(testName, rg.Name)
+                    .ConfigureAwait(false);
+
+                // check the name is now available
+                availabilityInfo = await _provisioningClient.IotDpsResource
+                    .CheckProvisioningServiceNameAvailabilityAsync(testName)
+                    .ConfigureAwait(false);
+                availabilityInfo.NameAvailable.Should().BeTrue();
+            }
 
-                if (!availabilityInfo.NameAvailable ?? false)
+            // try to create a DPS service
+            var createServiceDescription = new ProvisioningServiceDescription(
+                Constants.DefaultLocation,
+                new IotDpsPropertiesDescription(),
+                new IotDpsSkuInfo(
+                    Constants.DefaultSku.Name,
+                    Constants.DefaultSku.Tier,
+                    Constants.DefaultSku.Capacity));
+
+            var dpsInstance = await _provisioningClient.IotDpsResource
+                .CreateOrUpdateAsync(
+                    rg.Name,
+                    testName,
+                    createServiceDescription)
+                .ConfigureAwait(false);
+
+            dpsInstance.Should().NotBeNull();
+            dpsInstance.Sku.Name.Should().Be(Constants.DefaultSku.Name);
+            dpsInstance.Name.Should().Be(testName);
+
+            // verify item exists in list by resource group
+            IPage<ProvisioningServiceDescription> existingServices = await _provisioningClient.IotDpsResource
+                .ListByResourceGroupAsync(rg.Name)
+                .ConfigureAwait(false);
+            existingServices.Should().Contain(x => x.Name == testName);
+
+            // verify can find
+            ProvisioningServiceDescription foundInstance = await _provisioningClient.IotDpsResource
+                .GetAsync(testName, testName)
+                .ConfigureAwait(false);
+            foundInstance.Should().NotBeNull();
+            foundInstance.Name.Should().Be(testName);
+
+            var attempts = Constants.ArmAttemptLimit;
+            var hasSucceeded = false;
+            while (attempts > 0 && !hasSucceeded)
+            {
+                try
                 {
-                    //it exists, so test the delete
-                    this.provisioningClient.IotDpsResource.Delete(testName, testName);
-
-                    //check the name is now available
-                    availabilityInfo =
-                        this.provisioningClient.IotDpsResource.CheckProvisioningServiceNameAvailability(new OperationInputs(testName));
-                    Assert.True(availabilityInfo.NameAvailable);
+                    await _provisioningClient.IotDpsResource
+                        .DeleteAsync(testName, rg.Name)
```

Comment:
I wonder why rg is second in this API, where I'm always used to seeing it be the first parameter for control-plane operations.

Reply:
The swagger is in this order. https://github.com/Azure/azure-rest-api-specs/blob/ca05b2795cf5ebafcd001682e6a0157626b9d2b8/specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2020-03-01/iotdps.json#L563

Reply:
Bummer. Too bad it has already GA'd